### PR TITLE
Rename APIs and add documentation

### PR DESCRIPTION
**Pull request by eproxus**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,18 +8,24 @@ and this project adheres to
 
 ## Unreleased
 
-### Changed
-
-- `pmod_gps` now validate the NMEA sentences and parse the GGA ones.
-- `pmod_gps` API changed, it now return a pre-parsed tuple, and for the GGA a parsed map of the sentence fields.
```
Comment (maehjam) [inline CHANGELOG.md:14]: The verbs miss the -s at the ending: validates, parses, returns.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,18 +8,24 @@ and this project adheres to
...
+- `grisp_rtems` module containing NIFs for selected RTEMS functions
 
 ### Changed
 
-- Replace SPI port driver with NIF
+- `grisp_gpio` interface has been updated and simplified
+- `grisp_spi` interface has been updated and simplified
+- `grisp_i2c` interface has been updated and simplified
+- `pmod_gps` now validate the NMEA sentences and parse the GGA ones
+- `pmod_gps` API changed, it now return a pre-parsed tuple, and for the GGA a parsed map of the sentence fields
```
Comment (maehjam) [inline CHANGELOG.md:22]: The verbs miss the -s ending: validates, parses, returns.

File: src/grisp_i2c.erl [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+-export_type([bus_name/0]).
+-export_type([bus/0]).
+-export_type([reg_addr/0]).
+-export_type([target_addr/0]).
+-export_type([message/0]).
+
+%--- API -----------------------------------------------------------------------
+
+% @doc List I²C buses.
```
Comment (maehjam) [inline src/grisp_i2c.erl:66]: I think there should be an -s at the ending of list: "Lists I²C buses".

File: src/grisp_i2c.erl [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+%   i2c1 => #{name => i2c1,path => <<"/dev/i2c-1">>}}
+% '''
+-spec buses() -> #{bus_name() := #{name := bus_name(), path := bus_path()}}.
+buses() ->
+    #{
+        i2c0 => #{name => i2c0, path => <<"/dev/i2c-0">>},
+        i2c1 => #{name => i2c1, path => <<"/dev/i2c-1">>}
+    }.
+
+% @doc Open an I²C bus device by name.
```
Comment (maehjam) [inline src/grisp_i2c.erl:81]: Ending -s again: Opens an...

File: src/grisp_i2c.erl [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+% 2> I2C0 = grisp_i2c:open(i2c1).
+% #Ref<0.4157010815.3886678017.238942>
+% '''
+% @see buses/0
+-spec open(bus_name()) -> bus().
+open(Name) ->
+    #{path := Path} = maps:get(Name, buses()),
+    i2c_open_nif(null(Path)).
+
+% @doc Detect I²C devices on a bus.
```
Comment (maehjam) [inline src/grisp_i2c.erl:94]: Ending -s: Detects ...

File: src/grisp_rtems.erl [hunk 6]
```diff
@@ -70,6 +86,35 @@ write_file_to_device(FilePath, DevicePath, ReadChunkSize, WriteChunkSize) ->
             Error
     end.
 
+% @doc Perform a raw `pwrite' syscall to a device.
```
Comment (maehjam) [inline src/grisp_rtems.erl:89]: Missing -s at the end of perform: Performs a raw...

File: src/grisp_spi.erl [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+% before the controller has finished sending a request, and even reply with
+% data while the controller is writing its own data.
+%
+% This module provides two message formats. One format ({@link message_raw
+% ()}) returns the full response, all bytes received from the responder device
+% including ones sent during the initial request.
+%
+% The other format ({@link message_simple()}) automatically pads the request so
+% that response bytes can be received, and strips the response of an initial
+% number of bytes. This is makes it simple to use responder devices which wait
```
Comment (maehjam) [inline src/grisp_spi.erl:57]: Remove "is": This makes it simple...

Comment (maehjam) [reply src/grisp_spi.erl:57]: I think "which" should be replaced with "that", because it's a defining clause.

File: src/grisp_spi.erl [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+% Chip select pin `CS' can be any valid GPIO output pin.
+%
+% @see grisp_gpio
+-spec open(bus(), grisp_gpio:pin()) -> ref().
+open(_Bus, CS) -> {open_nif(), pin(grisp_hw:platform(), CS)}.
+
+% @doc Transfer SPI messages on a bus.
+%
+% A message can either be a raw binary message ({@link message_raw()}), for
+% which a response of the same length is returned. Alternatively, it can be a
```
Comment (maehjam) [inline src/grisp_spi.erl:134]: The "either ... Alternatively,..." structure is messing up my reading flow. I am looking for the "or" from "either .. or".

**Review by maehjam — CHANGES_REQUESTED** [review]
We need some consistent grammar for function documentation. I've seen the description of a function usually does not have a subject, starts with a verb in 3rd person singular and ends with a period. One can interpret the function name to be the subject and this way it's fine for the first sentence. Additionally, since it's 3rd person singular the verbs should have an ending -s, I did mark some of them, but not all. Besides those descriptive first sentences, I would prefer complete sentences with subject, verb and object.

**Comment by eproxus** [comment]
Agree. Erlang/OTP seems to mostly use the simple present passive form, e.g.
> Returns a list with the elements in `List1` in reverse order.

If we are okay with this form, I'll update the documented functions affected by this PR to match this pattern.

File: src/grisp_spi.erl (hunk 8 above)
Comment (eproxus) [reply src/grisp_spi.erl]: Please check the new text, I changed it around a bit.

File: src/grisp_i2c.erl [hunk 9]
```diff
@@ -0,0 +1,174 @@
+% @doc GRiSP I²C API.
+%
+% <a href="https://en.wikipedia.org/wiki/I²C">Inter-Integrated Circuit
+% (I²C)</a> is a synchronous addressable serial communication bus. Two types of
+% nodes can exist, controllers and targets. Multiple controllers and targets
+% can exist on the bus simultaneously. A GRiSP board acts as a controller to
+% any connected target peripherals. The controller generate the clock signal
+% and initiate communication with targets. Targets follow the clock and
+% respond when addressed.
```
Comment (maehjam) [inline src/grisp_i2c.erl:9]: There is a missing -s at the end of controller, it's plural now, isn't it? Hence, I would write:
> Controllers generate...

File: src/grisp_i2c.erl [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+% bus talks to any connected target peripherals, either on the PMOD connector
+% or the raised I²C pins.
+%
+% Communication consists of either read or write messages. A read or write must
+% be addressed to a specific target (chip). A target is addressed by a 7-bit
+% address. The registry to use is specified in a target specific way. E.g. to
+% read a register some targets expect a write message with the register address
+% as a value, followed by a read message.
+%
+% <em>Please refer to to the specification for a specific target chip for
```
Comment (maehjam) [inline src/grisp_i2c.erl:23]: > to to

File: src/grisp_i2c.erl [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+% 3> [io_lib:format("0x~.16B", [Target]) || Target <- grisp_i2c:detect(I2C0)].
+% ["0x18","0x36","0x37","0x52","0x57","0x5A","0x5F"]
+% '''
+-spec detect(bus()) -> [target_addr()].
+detect(Bus) -> [Target || Target <- lists:seq(1, 127), present(Bus, Target)].
+
+% @doc Performs a simplified read from a register on an I²C chip.
+%
+% This function sends a write message with the single byte register address as
+% value, then a read message of of the specified length.
```
Comment (maehjam) [inline src/grisp_i2c.erl:109]: > of of

File: src/grisp_spi.erl [hunk 12]
```diff
@@ -0,0 +1,207 @@
+% @doc GRiSP SPI API.
+%
+% <a href="https://en.wikipedia.org/wiki/Serial_Peripheral_Interface">Serial
+% Peripheral Interface (SPI)</a> is a synchronous serial communication
+% protocol, where a controller device controls one to many responder devices.
```
Comment (maehjam) [inline src/grisp_spi.erl:5]: > one to many

This sounds like "one too many". I would rephrase to "one or several".

File: src/grisp_i2c.erl (hunk 9 above)
Comment (eproxus) [reply src/grisp_i2c.erl:9]: Ah, misinterpreted your suggestion.

File: src/grisp_i2c.erl (hunk 11 above)
Comment (eproxus) [reply src/grisp_i2c.erl:109]: 🤦

File: src/grisp_i2c.erl (hunk 10 above)
Comment (eproxus) [reply src/grisp_i2c.erl:23]: 🤦

**Review by maehjam — APPROVED** [review]
